sdk/storage/azblob/blockblob/client_test.go

```diff
@@ -587,6 +587,674 @@ func (s *BlockBlobRecordedTestsSuite) TestUploadBlockWithImmutabilityPolicy() {
+	_require.NoError(err)
+
+	containerClient, _, destBlob, srcBlobURLWithSAS, _ := setUpPutBlobFromURLTest(testName, _require, svcClient)
+	defer testcommon.DeleteContainer(context.Background(), _require, containerClient)
+
+	// Invoke UploadBlobFromURL
+	tier := blob.AccessTierCool
+	options := blockblob.UploadBlobFromURLOptions{
+		Tags:        testcommon.BasicBlobTagsMap,
+		HTTPHeaders: &testcommon.BasicHeaders,
+		Metadata:    testcommon.BasicMetadata,
+		Tier:        &tier,
+	}
+
+	pbResp, err := destBlob.UploadBlobFromURL(context.Background(), srcBlobURLWithSAS, &options)
+	_require.NotNil(pbResp)
+	_require.NoError(err)
+
+	// Check dest and source properties
+	resp, err := destBlob.GetProperties(context.Background(), nil)
+	_require.NoError(err)
+	h := blob.ParseHTTPHeaders(resp)
+	h.BlobContentMD5 = nil // the service generates a MD5 value, omit before comparing
+	_require.EqualValues(h, testcommon.BasicHeaders)
+	_require.EqualValues(resp.AccessTier, &tier)
+	tagcount := int64(len(testcommon.BasicBlobTagsMap))
+	_require.EqualValues(resp.TagCount, &tagcount)
+	_require.EqualValues(resp.Metadata, testcommon.BasicMetadata)
+}
+
+func (s *BlockBlobUnrecordedTestsSuite) TestPutBlobFromUrlWithCPK() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	_require.NoError(err)
+
+	containerClient, _, destBlob, srcBlobURLWithSAS, _ := setUpPutBlobFromURLTest(testName, _require, svcClient)
+	defer testcommon.DeleteContainer(context.Background(), _require, containerClient)
+
+	// Invoke UploadBlobFromURL
+	options := blockblob.UploadBlobFromURLOptions{
+		CPKInfo: &testcommon.TestCPKByValue,
+	}
+
+	pbResp, err := destBlob.UploadBlobFromURL(context.Background(), srcBlobURLWithSAS, &options)
+	_require.NotNil(pbResp)
+	_require.NoError(err)
+
+	// Get CPKInfo and compare
+	getBlobPropertiesOptions := blob.GetPropertiesOptions{
+		CPKInfo: &testcommon.TestCPKByValue,
+	}
+
+	getResp, err := destBlob.GetProperties(context.Background(), &getBlobPropertiesOptions)
+	_require.Nil(err)
+	_require.EqualValues(getResp.EncryptionKeySHA256, testcommon.TestCPKByValue.EncryptionKeySHA256)
+}
+
+func (s *BlockBlobUnrecordedTestsSuite) TestPutBlobFromUrlCPKScope() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	encryptionScope := testcommon.GetCPKScopeInfo(s.T())
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	_require.NoError(err)
+
+	containerClient, _, destBlob, _, _ := setUpPutBlobFromURLTest(testName, _require, svcClient)
+	defer testcommon.DeleteContainer(context.Background(), _require, containerClient)
+
+	// Create Blob with CPK
+	bbName := testcommon.GenerateBlobName(testName)
+	srcBlob := testcommon.CreateNewBlockBlobWithCPK(context.Background(), _require, bbName, containerClient, nil, &encryptionScope)
+	expiryTime, err := time.Parse(time.UnixDate, "Fri Jun 11 20:00:00 UTC 2049")
```

Comment:
Same as previous for this as well as the error check after `testcommon.GetGenericSharedKeyCredential`.

Reply:
Updated!